### PR TITLE
fix: add safekeeper team to pgxn codeowners

### PR DESCRIPTION
**Pull request by skyzh**
## Problem

`pgxn/` also contains WAL proposer code, so modifications to this directory should be able to be approved by the safekeeper team.

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics? if so did you add the relevant metrics to the dashboard?
- [ ] If this PR requires public announcement, mark it with /release-notes label and add several sentences in this section.

## Checklist before merging

- [ ] Do not forget to reformat commit message to not include the above checklist

### CODE REVIEW

**Comment by skyzh** [comment]
e.g., https://github.com/neondatabase/neon/commits/main/pgxn/neon/walproposer.c is part of the safekeeper

**Comment by jcsp** [comment]
@arssher wdyt?  does it make sense to require compute team review for changes to the C code that we link into postgres, or would you like your team to be able to do these changes unilaterally?

File: CODEOWNERS
Comment (petuhovskiy) [inline CODEOWNERS:9]: /pgxn/neon/ is a better option. And it would be even better if we can require approval only for `walproposer*` files.

**Comment by arssher** [comment]
> wdyt? does it make sense to require compute team review for changes to the C code that we link into postgres, or would you like your team to be able to do these changes unilaterally?

We definitely do want to be able to make them unilaterally; if this is not the case, I just manually ask for the review.

> The other side effect of this change would be that your team would get auto-assigned as reviewers for pgxn code that is unrelated

True, but there is no easy ideal solution. Even without it generally in my experience this codeowners thing quite frequently assigns unsuitable reviewers, but previous point (being able to merge without unrelated team reviews) is more important.

**Comment by arssher** [comment]
We should also add @neondatabase/safekeepers to /libs/postgres_ffi/. In fact likely this PR is caused by
https://github.com/neondatabase/neon/pull/7143

**Review by arssher — APPROVED** [review]

File: CODEOWNERS
Comment (skyzh) [reply CODEOWNERS]: resolved in https://github.com/neondatabase/neon/pull/7170/commits/c38db0bc8fe8391f55dd1be84bdfbd36072bf638

Comment (skyzh) [reply CODEOWNERS]: I only added `pgxn/neon/` so that team-safekeeper can have the flexibility of adding new directories/files.

**Pull request merged**
